utils/disk_space_monitor.hh

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <chrono>
+#include <filesystem>
+#include <set>
+
+#include <boost/intrusive/list.hpp>
+
+#include <seastar/core/gate.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/timer.hh>
+#include <seastar/util/optimized_optional.hh>
+
+#include "seastarx.hh"
+
+using namespace std::chrono_literals;
+
+namespace bi = boost::intrusive;
+
+namespace utils {
+
+// Instantiated only on shard 0
+class disk_space_monitor {
+public:
+    struct config {
+        scheduling_group sched_group;
+        lowres_clock::duration normal_polling_interval = 10s;
+        lowres_clock::duration high_polling_interval = 1s;
+        // Use high_polling_interval above this thrshold
```

Comment:
Typo: `threshold`

Reply:
Fixed in [8616db6](https://github.com/scylladb/scylladb/commit/8616db640cbca07d42990522d95f3b2983c3197c)